store/tikv/lock_resolver.go

```diff
@@ -578,10 +595,136 @@ func (lr *LockResolver) getTxnStatus(bo *Backoffer, txnID uint64, primary []byte
 			status.commitTS = cmdResp.CommitVersion
 			lr.saveResolved(txnID, status)
 		}
+
 		return status, nil
 	}
 }
 
+// resolveLockAsync resolves l assuming it was locked using the async commit protocol.
+func (lr *LockResolver) resolveLockAsync(bo *Backoffer, l *Lock, status TxnStatus, lite bool, cleanRegions map[RegionVerID]struct{}) error {
+	tikvLockResolverCountWithResolveAsync.Inc()
+
+	regions, _, err := lr.store.GetRegionCache().GroupKeysByRegion(bo, status.secondaries, nil)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	commitTs := l.MinCommitTS
+	commitTsMutex := sync.Mutex{}
+	locks := []*Lock{l}
+	missingLock := false
+	errChan := make(chan error, len(regions))
+
+	for regionID, keys := range regions {
+		curRegionID := regionID
+		curKeys := keys
+
+		go func() {
+			for {
+				checkReq := &kvrpcpb.CheckSecondaryLocksRequest{
+					Keys:         curKeys,
+					StartVersion: l.TxnID,
+				}
+				req := tikvrpc.NewRequest(tikvrpc.CmdCheckSecondaryLocks, checkReq)
+				tikvLockResolverCountWithQueryCheckSecondaryLocks.Inc()
+				resp, err := lr.store.SendReq(bo, req, curRegionID, readTimeoutShort)
+				if err != nil {
+					errChan <- errors.Trace(err)
+					return
+				}
+				regionErr, err := resp.GetRegionError()
+				if err != nil {
+					errChan <- errors.Trace(err)
+					return
+				}
+				if regionErr != nil {
+					err = bo.Backoff(BoRegionMiss, errors.New(regionErr.String()))
+					if err != nil {
+						errChan <- errors.Trace(err)
+						return
+					}
+					continue
```

Comment:
I just realize that maybe it's not correct to just `continue`. If the region splits or merges, `curKeys` may be not all in `curRegionID` and then the request will never succeed.
So I think `curKeys` should be grouped by region again on a region miss.

Reply:
Why is this OK for a regular resolve lock, but not with async commit?